### ISSUE TITLE
508-defect-2 [SEMANTIC MARKUP, COGNITION]: Anticipated Separation: <legend> SHOULD be announced to screen readers when in form groupings 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19013]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I need to know that a range of form inputs are grouped together, so that I may understand the scope of the use case.

## Details

Groups of form elements **MUST** have group labels (e.g. `<fieldset>` and `<legend>`, to meet Section 508 requirements. The current code for Anticipated Separation _has_ a `<legend>`, but it does not:
- Wrap the input field "Enter a location"
- Is not a first child within the `<fieldset>`

Wrapping the input field "Enter a location" within the `<fieldset>` will provide the legend "Anticipated Separation" upon tabbing into the input field "Enter a location" which helps users understand _what_ location they need to provide. If the `<legend>` is _not_ the first child of the `<fieldset>`, however, it may not be announced on certain screen reader browser combinations.

I've written this as a defect-2 since a legend already exists and there aren't multiple fieldsets 🙇 

## Acceptance Criteria

- [x] Wrap the input field within the `fieldset`
- [ ] Make the legend as a direct child under `fieldset`

### Current code (Simplified)

```html
<fieldset>
  <div>
    <div>
      <legend>
        Date or anticipated date of release from active duty
      </legend>
      <!-- Accordion hint -->
    </div>
  </div>
<fieldset>
<!-- Input field -->
```

### Recommended code (Simplified)

```html
<fieldset>
  <legend>
        Date or anticipated date of release from active duty
  </legend>
  <div>
    <div>
      <!-- Accordion hint -->
      <!-- Input field -->
    </div>
  </div>
<fieldset>
```

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/disability/file-disability-claim-form-21-526ez/introduction ` in browser
2. Continue to Anticipated Separation page
3. Start screen reader 
4. Tab to "What's this?" and confirm the legend is not announced
5. Tab to input field and confirm the legend is not announced

### Current code (Simplified)

```html
<fieldset>
  <div>
    <div>
      <legend>
        Date or anticipated date of release from active duty
      </legend>
      <!-- Accordion hint -->
    </div>
  </div>
<fieldset>
<!-- Input field -->
```

### Recommended code (Simplified)

```html
<fieldset>
  <legend>
        Date or anticipated date of release from active duty
  </legend>
  <div>
    <div>
      <!-- Accordion hint -->
      <!-- Input field -->
    </div>
  </div>
<fieldset>
```

## WCAG or Vendor Guidance (optional)

* [WCAG 2.0 Level A - 1.3.1 - Info and Relationships: Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

* [Wuhcag 1.3.1 – Info and Relationships (Level A)](https://www.wuhcag.com/info-and-relationships/)

* [WCAG 2.0 Level A - 4.1.1 - Parsing](http://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-parses.html)

## Screenshots or Trace Logs
<img width="1438" alt="Screen Shot 2021-01-26 at 9 36 00 AM" src="https://user-images.githubusercontent.com/14154792/105859627-c0e94180-5fba-11eb-9fd5-953fc4d493fc.png">

**Comment by Mottie** [comment]
Fix merged into staging - it should be available soon

**Comment by joshkimux** [comment]
@Mottie Brilliant! Looks like the legend as first child issue is all that's left-- but from my memory I think that's form system related. I updated the acceptance criteria to match the changes you made 😄

**Comment by Mottie** [comment]
Yes, the form system adds all those nested divs... it'll be difficult to remove them since they have class names that may, or may not (I didn't look) on styling.

**Comment by joshkimux** [comment]
Let's keep the ticket open for now! Hopefully a fix will come down the road for the form system 🤞

**Comment by caw310** [comment]
@SarahKay8 , this needs to be retested. I will prioritize this for an upcoming sprint.

**Comment by SarahKay8** [comment]
Per Josh's recommendations, input field "Enter a location" has been wrapped within the <fieldset> 

See photo for reference: ![Image 8-16-22 at 11.51 AM.jpg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/af35a1b1-600b-443f-8be2-5328ae982019)

@caw310
